### PR TITLE
feat: [BREAKING] quote via messages; use connector names, not URIs; removeClient()

### PR DESCRIPTION
**Pull request by sentientwaffle**
Requires https://github.com/interledgerjs/ilp-routing/pull/12

Fixes https://github.com/interledgerjs/ilp-core/issues/49

### CODE REVIEW

File: src/lib/client.js [hunk 1]
```diff
@@ -8,14 +8,15 @@ const omitUndefined = require('lodash/fp/omitBy')(isUndefined)
...
+const debug = require('debug')('ilp-core')
 
 class Client extends EventEmitter {
   /**
    * @param {Object} pluginOpts options for the ledger plugin
    * @param {Function} pluginOpts._plugin A ledger plugin constructor
    * @param {Object} [_clientOpts]
-   * @param {URI[]} [_clientOpts.connectors] A list of connectors to quote from
+   * @param {String[]} [_clientOpts.connectors] A list of connectors to quote from
+   * @param {Integer} [_clientOpts.messageTimeout] The number of milliseconds to wait for a response to sendMessage.
```
Comment (emschwartz) [inline src/lib/client.js]: If this is an unreliable message anyway, why should it bother waiting?

File: src/lib/core.js [hunk 2]
```diff
@@ -18,7 +17,12 @@ class Core extends EventEmitter {
     super()
     this.clientList = [] // Client[]
     this.clients = {} // { prefix ⇒ Client }
-    this.tables = options.routingTables || new RoutingTables(null, [], null)
+    this.tables = options.routingTables || new RoutingTables([], null)
+
+    const core = this
+    this._relayEvent = function (event, arg1, arg2, arg3) {
+      return core.emitAsync(event, this, arg1, arg2, arg3)
```
Comment (emschwartz) [inline src/lib/core.js]: I think this should be done with the `arguments` keyword instead of picking a certain number of args

Comment (emschwartz) [inline src/lib/core.js]: Also wouldn't you want to know what client it came from?

Comment (sentientwaffle) [inline src/lib/core.js]: re: "what client it came from" that's what `this` is inside the function's context.

File: src/lib/client.js (hunk 1 above)
Comment (sentientwaffle) [inline src/lib/client.js]: The timeout isn't for the `POST /messages`; it's how long to wait for a response to a `quote_request` (or any other message that expects another message to be sent back in reply).

**Comment by sentientwaffle** [comment]
@emschwartz updated per your review

**Pull request merged**